File: C atau C++/programatm.cpp
#include <iostream>
#include <string>
#include <iomanip>
using namespace std;

string rupiah(int angka) {
    string s = to_string(angka);
    string hasil = "Rp ";
    int hitung = 0;

    for (int i = s.size() - 1; i >= 0; i--) {
        hasil = s[i] + hasil;
        hitung++;
        if (hitung == 3 && i != 0) {
            hasil = "." + hasil;
            hitung = 0;
        }
    }
    return hasil;
}

int main() {
    string nama, confirm;
    long long nim, saldo;

    do {
        cout << "Masukkan nama: ";
        getline(cin, nama);

        cout << "Apakah nama sudah benar? (true/false): ";
        getline(cin, confirm);

    } while (confirm != "true");

    cout << "Masukkan NIM tanpa huruf: ";
    cin >> nim;
    saldo = nim;

    cout << "\n==== DATA NASABAH ====\n";
    cout << "Nama : " << nama << endl;
    cout << "Saldo Awal : " << rupiah(saldo) << "\n\n";

    int menu;

    while (true) {
        cout << "===== MENU ATM =====\n"
             << "1. Cek Saldo\n"
             << "2. Tarik Tunai\n"
             << "3. Setor Tunai\n"
             << "4. Transfer\n"
             << "5. Keluar\n"
             << "Pilih Menu : ";
        cin >> menu;

        switch (menu) {
            case 1: 
                cout << "Saldo kamu: " << rupiah(saldo) << "\n\n";
                break;

            case 2: { 
                int tarik;
                cout << "Jumlah tarik: ";
                cin >> tarik;
                if (tarik <= saldo) {
                    saldo -= tarik;
                    cout << "Berhasil tarik! Sisa saldo: " << rupiah(saldo) << "\n\n";
                } else {
                    cout << "Saldo tidak cukup!\n\n";
                }
                break;
            }

            case 3: { 
                int setor;
                cout << "Jumlah setor: ";
                cin >> setor;
                saldo += setor;
                cout << "Saldo sekarang: " << rupiah(saldo) << "\n\n";
                break;
            }

            case 4: { 
                int tf;
                cout << "Jumlah transfer: ";
                cin >> tf;
                if (tf <= saldo) {
                    saldo -= tf;
                    cout << "Transfer berhasil! Sisa saldo: " << rupiah(saldo) << "\n\n";
                } else {
                    cout << "Saldo tidak cukup!\n\n";
                }
                break;
            }

            case 5: 
                cout << "Terima kasih sudah menggunakan ATM!\n";
                return 0;

            default:
                cout << "Menu tidak tersedia!\n\n";
        }
    }
}

File: php/programatm.php
<?php

do {
    echo "Masukkan nama : ";
    $nama = trim(fgets(STDIN));

    echo "Apakah nama sudah benar? (true/false) : ";
    $confirm = strtolower(trim(fgets(STDIN)));

} while ($confirm !== "true");

echo "Masukkan NIM tanpa huruf : ";
$nim = trim(fgets(STDIN));
$saldo = (int)$nim;

echo "\n====  DATA NASABAH  ====\n";
echo "Nama : $nama\n";
echo "Saldo Awal : Rp " . number_format($saldo, 0, ',', '.') . "\n";


while (true) {
    echo "\n====  MENU ATM  ====\n";
    echo "1. Cek saldo\n";
    echo "2. Tarik Tunai\n";
    echo "3. Setor Tunai\n";
    echo "4. Transfer\n";
    echo "5. Keluar\n";
    echo "Pilih Menu : ";
    $menu = (int)trim(fgets(STDIN));

    switch ($menu) {

        case 1:
            echo "Saldo kamu: Rp " . number_format($saldo, 0, ',', '.') . "\n";
            break;

        case 2:
            echo "Jumlah tarik : ";
            $tarik = (int)trim(fgets(STDIN));
            if ($tarik <= $saldo) {
                $saldo -= $tarik;
                echo "Berhasil tarik! Sisa saldo: Rp " . number_format($saldo, 0, ',', '.') . "\n";
            } else {
                echo "Saldo tidak cukup!\n";
            }
            break;

        case 3:
            echo "Jumlah setor : ";
            $setor = (int)trim(fgets(STDIN));
            $saldo += $setor;
            echo "Saldo sekarang: Rp " . number_format($saldo, 0, ',', '.') . "\n";
            break;

        case 4:
            echo "Jumlah transfer : ";
            $tf = (int)trim(fgets(STDIN));
            if ($tf <= $saldo) {
                $saldo -= $tf;
                echo "Transfer berhasil! Sisa saldo: Rp " . number_format($saldo, 0, ',', '.') . "\n";
            } else {
                echo "Saldo tidak cukup!\n";
            }
            break;

        case 5:
            echo "Terimakasih sudah menggunakan ATM\n";
            exit;

        default:
            echo "Menu tidak tersedia!\n";
    }
}
?>